include/hal/gpio_event_controller.hpp

```diff
@@ -0,0 +1,46 @@
+#include "hal/hal_enums.hpp"
+
+#include <cstdint>
+
+namespace edge::aidan {
+
+extern "C" void GPIOTE_IRQHandler();
+
+static constexpr uint8_t GPIO_PINS = 32;
+
+// TODO: Add a controller for other GPIOTE events (channels)?
+// Avoiding premature optimization, lockign it to just pin events for now
```

Comment:
this is not premature optimization lmao...

Reply:
like fine to leave a TODO bc its scope but

Reply:
this comment should've been under gpiote handler (the only direct coupling)

updated gpiote handler to reflect that this is literally just a matter of if statements